### PR TITLE
[WIP] Allow downloading of patches from github.

### PR DESCRIPTION
**Pull request by groutr**
This PR adds a feature to conda to allow downloading pull requests from github as patches which will then be applied to the source tree.

There are a couple of approaches
1) Download patch to a temporary file and apply to source tree
2) Download patch to a regular file in recipe directory and apply to source tree.  This would make the builds more reproducible in case the PR went away (in which case, the patch should be re-examined).

TODO:
- [ ] Add error checking code.  This assume that the correct github url has been given.  If there is an error, it will most likely die.
- [ ] Add tests (when idea is finalized).
- [ ] Start discussion on how to best implement this.

### CODE REVIEW

File: conda_build/source.py [hunk 1]
```diff
@@ -219,10 +219,20 @@ def parse_bool(s):
...
+    patch -p1 patches.
+
+    All local patches (in the recipe directory) are assumed to be
+    patch -p0 patches
+    """
     print('Applying patch: %r' % path)
     if not isfile(path):
         sys.exit('Error: no such patch: %s' % path)
-
+
```
Comment (hargup) [inline conda_build/source.py]: There are trailing whitespaces.

**Comment by rmcgibbo** [comment]
Really missing this feature. It would be very convenient to have.

**Comment by msarahan** [comment]
@groutr - any interest in revisiting this one?

**Comment by groutr** [comment]
@msarahan I can revisit this in my spare time.  Have we already implemented the changes in https://github.com/conda/conda-build/pull/1006?

**Comment by msarahan** [comment]
Nope.  Not sure where that stands.  Ilan asked that we make a separate package instead.

**Comment by beeankha** [comment]
This pull request is being closed in order to reduce the number of outdated/stale PRs on this repo; if this work is still actively in progress, please feel free to re-open!

**Pull request closed without merging**